pkg/resources/ingresses/ingress.go

```diff
@@ -57,16 +50,24 @@ func NewIngressForCRWithSSL(namespacedName types.NamespacedName, labels map[stri
 			},
 		},
 	}
+
+	portName := ""
+	portNumber := -1
+	portValue := intstr.FromString(targetPortName)
+	if portNumber = portValue.IntValue(); portNumber == 0 {
+		portName = portValue.String()
+	}
+
 	if portName == "" {
-		ingress.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port = netv1.ServiceBackendPort{
+		desired.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port = netv1.ServiceBackendPort{
 			Number: int32(portNumber),
 		}
 	} else {
-		ingress.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port = netv1.ServiceBackendPort{
+		desired.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port = netv1.ServiceBackendPort{
 			Name: portName,
 		}
 	}
-	ingress.Spec.Rules[0].Host = "www.mgmtconsole.com"
+	desired.Spec.Rules[0].Host = "www.mgmtconsole.com"
 	if sslEnabled {
 		//ingress assumes TLS termination at the ingress point and uses default https port 443
 		//it requires a host name that matches the certificate's CN value
```

Comment:
```suggestion
	host := desired.GetObjectMeta().GetName() + ".apps.artemiscloud.io"
	desired.Spec.Rules[0].Host = host
	if sslEnabled {
		//ingress assumes TLS termination at the ingress point and uses default https port 443
		//it requires a host name that matches the certificate's CN value
		//it also uses a self-signed cert if you don't provide one
		//this is the default configuration in minikube
		tls := []netv1.IngressTLS{
			{
				Hosts: []string{host},
```

Reply:
looks good. does anything currently depend on that hard coded mgmtconsole.com doman name?